### DISCUSSION TITLE
Unable to get rose-pine to work

### DISCUSSION THREAD

**Discussion by psomdeb25** [rose-pine/neovim#244] [Q&A] [answered]
I copied and pasted this (enclosed in code block below) entire section from the README file into my init.vim file but it does not apply to my NeoViM configuration:

```nvim
    require("rose-pine").setup({
    variant = "auto", -- auto, main, moon, or dawn
    dark_variant = "main", -- main, moon, or dawn
    dim_inactive_windows = false,
    extend_background_behind_borders = true,

    enable = {
        terminal = true,
        legacy_highlights = true, -- Improve compatibility for previous versions of Neovim
        migrations = true, -- Handle deprecated options automatically
    },

    styles = {
        bold = true,
        italic = true,
        transparency = false,
    },

    groups = {
        border = "muted",
        link = "iris",
        panel = "surface",

        error = "love",
        hint = "iris",
        info = "foam",
        note = "pine",
        todo = "rose",
        warn = "gold",

        git_add = "foam",
        git_change = "rose",
        git_delete = "love",
        git_dirty = "rose",
        git_ignore = "muted",
        git_merge = "iris",
        git_rename = "pine",
        git_stage = "iris",
        git_text = "rose",
        git_untracked = "subtle",

        h1 = "iris",
        h2 = "foam",
        h3 = "rose",
        h4 = "gold",
        h5 = "pine",
        h6 = "foam",
    },

    highlight_groups = {
        -- Comment = { fg = "foam" },
        -- VertSplit = { fg = "muted", bg = "muted" },
    },

    before_highlight = function(group, highlight, palette)
        -- Disable all undercurls
        -- if highlight.undercurl then
        --     highlight.undercurl = false
        -- end
        --
        -- Change palette colour
        -- if highlight.fg == palette.pine then
        --     highlight.fg = palette.foam
        -- end
    end,
})

vim.cmd("colorscheme rose-pine")
```

This is the following error that I get:
`not an editor command require("rose-pine")`

Is there another "Preconfiguration" that I need to do?

Thank you very much!

**Comment by mvllow** [comment, marked as answer]
You'll have to follow the installation instructions as well. Depending on your package manager, add "rose-pine/neovim" alongside your other plugins. If that doesn't sound familiar, you could use the builtin package manager like so:

```sh
# In your terminal
mkdir -p ~/.local/share/nvim/site/pack/user/start
cd ~/.local/share/nvim/site/pack/user/start
git clone https://github.com/rose-pine/neovim.git
```

Edit: You mentioned "init.vim" as well. If you are using vimscript, wrap the code above:

```vimscript
lua << EOF
require("rose-pine").setup()
vim.cmd("colorscheme rose-pine")
EOF
```

**Comment by psomdeb25** [comment]
Thank you for the answer! Your EDIT actually worked for me. 

I am familiar installing packages in neovim using Vundle. I had installed using the `:VundleInstall` command. In command mode when I typed in `:colorscheme ` and pressed the `TAB` key, in the context menu the package was also listed.

I believe what you mentioned in the EDIT about enclosing within "lua << EOF ..... EOF" is essential for NeoVim users. Is there any particular reason why this is the case?

I can understand if this is not an appropriate place to ask this question. If you don't want to answer it, it's perfectly alright.

Thank you anyway for the providing the solution to my original question!

**Reply by mvllow** [reply]
Not a problem. This repository is written in lua and therefore assumes (for better or worse) that the user is also using lua. You are able to have an init.lua instead of init.vim and in that case you would not need the extra syntax mentioned above. The extra syntax allows writing lua code in your ".vim" files. Depending on how you plan on configuring Neovim, it could be beneficial to [learn more about using lua in Neovim](https://neovim.io/doc/user/lua-guide.html) but it's not necessary.

**Reply by psomdeb25** [reply]
Thank you for the answer! I think I should learn more about Lua, that a lot of NeoVim configurations are done using the language.
